utils/ctimer.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef UTILS_CTIMER_H_
+#define UTILS_CTIMER_H_
```

Comment:
Why would you call it `ctimer`?

Reply:
I have no idea what name it should have, but originally I think it is semantically equal as `callback timer` or `callback timeout`. Or should we just call it `timeout`?

Reply:
Various timer implementations come with callback functions. It is common, and you don't have to specify with names.